common/lib-dohttp/do_cpprest_base_uri.h

```diff
@@ -0,0 +1,394 @@
+#ifndef _DELIVERY_OPTIMIZATION_DO_BASE_URI_H
+#define _DELIVERY_OPTIMIZATION_DO_BASE_URI_H
+
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+#include "do_cpprest_utils.h"
+
+namespace microsoft
+{
+namespace deliveryoptimization
+{
+namespace details
+{
+namespace cpprest_web
+{
+
+/*
+    Code borrowed from cpprestsdk project: https://github.com/microsoft/cpprestsdk/
```

Comment:
>Code borrowed [](http://example.com/codeflow?start=4&length=13)

@jimson-msft I've copied the code as is for the most part. Only differences are in the types (no support for wide strings) and namespaces.